### ISSUE TITLE
[Bug]: Nested CSS warning after updating to latest release

### ISSUE THREAD

**Issue by endigo9740** [storybookjs/storybook#27161]
### Describe the bug

I'm receiving the following warnings after updating to the `@latest` release:

![Screenshot 2024-05-15 at 9 35 14 AM](https://github.com/storybookjs/storybook/assets/1509726/725b306b-b44a-427b-93e6-02dc4c0d89ae)

These styles are not present within my project, and we go out of our way to avoid using nested CSS.

![Screenshot 2024-05-15 at 9 57 43 AM](https://github.com/storybookjs/storybook/assets/1509726/2301066f-ef5d-4a8d-a018-b101b362821d)

### To Reproduce

Migrate to the `@latest` release as of today.

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.4
    CPU: (8) arm64 Apple M1
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 10.2.0 - /opt/homebrew/bin/npm
    pnpm: 8.10.2 - /opt/homebrew/bin/pnpm <----- active
  Browsers:
    Chrome: 124.0.6367.203
    Safari: 17.4
  npmPackages:
    @storybook/addon-essentials: ^8.1.1 => 8.1.1 
    @storybook/addon-interactions: ^8.1.1 => 8.1.1 
    @storybook/addon-links: ^8.1.1 => 8.1.1 
    @storybook/addon-svelte-csf: ^4.1.2 => 4.1.2 
    @storybook/addon-themes: ^8.1.1 => 8.1.1 
    @storybook/blocks: ^8.1.1 => 8.1.1 
    @storybook/preview-api: ^8.1.1 => 8.1.1 
    @storybook/svelte: ^8.1.1 => 8.1.1 
    @storybook/sveltekit: ^8.1.1 => 8.1.1 
    @storybook/test: ^8.1.1 => 8.1.1 
    eslint-plugin-storybook: ^0.8.0 => 0.8.0 
    storybook: ^8.1.1 => 8.1.1 
    storybook-dark-mode: ^4.0.1 => 4.0.1
```

### Additional context

The warning mentions arranging the order of Tailwind plugins. We do use Tailwind, but we do not use the Tailwind Nested CSS plugin. Again, we go out of our way to avoid nesting.

**Comment by onishi-kohei** [comment]
@endigo9740 
In a Nuxt project, I also encountered the same issue. I was able to suppress the warning by introducing tailwindcss/nesting on the project side.
```js
import tailwindcssNesting from 'tailwindcss/nesting'

const config = {
  vite: {
    css: {
      postcss: {
        plugins: [tailwindcssNesting]
      }
    }
  }
}

export default config
```

**Comment by endigo9740** [comment]
@onishi-kohei good to know. Unfortunately we would rather avoid the extra dependency.

**Comment by YannDuv** [comment]
Same warning, but with a Lit project. We are using tailwindCSS but no nesting CSS.

```bash
2:16:07 PM [vite] ✨ optimized dependencies changed. reloading
[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.
See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
125|      box-sizing: border-box;
126|  
127|      & ol {
   |      ^^^^^^
128|        padding-left: 18px;
   |  ^^^^^^^^^^^^^^^^^^^^^^^^^
129|        margin: 0;
   |  ^^^^^^^^^^^^^^^^
130|      }
   |  ^^^^^
```

**Comment by brycenaddison** [comment]
Same issue in React as well

**Comment by l0kt33n** [comment]
I updated the related packages to latest, and it seems to have fixed or suppress the nested CSS warning.

These are the package versions.
```json
    "postcss": "8.4.38",
    "tailwindcss": "3.4.3",
    "vite": "5.2.12",
```
postcss.config.js
```js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}

```

**Comment by pawelgalazka** [comment]
I can confirm that by updating to TailwindCSS `v3.4.3`, warning is not displayed anymore

**Comment by endigo9740** [comment]
Also confirming this is resolved with a Tailwind update to `v3.4.7`. With that I'll consider this issue resolve. Thanks for the help everyone!
